src/Sentry/Laravel/Tracing/CallableProxy.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sentry\Laravel\Tracing;
+
+use Illuminate\Contracts\Container\Container;
+use ReflectionClass;
+use ReflectionMethod;
+use Sentry\Tracing\SamplingContext;
+
+class CallableProxy
+{
+    /**
+     * @var Container
+     */
+    private $container;
+
+    public function __construct(Container $container)
+    {
+        $this->container = $container;
+    }
+
+    /**
+     * @param array|callable|null $tracesSampler
+     *
+     * @return array|callable
+     */
+    public function proxyTraceSampler($tracesSampler)
```

Comment:
Is there a way to make this more generic? So we can proxy any callable? I would like to be able to apply this to all possible callback enabled options and this is highly specific for the `traces_sampler` at the moment.

Reply:
That was my thought too. but the problem is the received parameter. Im gona try to prevent the container changing the "context"

Reply:
So, what I thought. We need to map the arguments with either name => $value, or class => $value. But I have not found a method that will return arguments of the function with `$variable` name. Depedency injection does not support `ordered` parameters.


```
if (is_callable($tracesSampler) === true) {
            return function () use ($tracesSampler) {
                return $this->container->call($tracesSampler, func_get_args());
            };
        }
```


We can do not bullet proof solution, but working solution:

-> loop all arguments and if it is a object, get the class and store it to a map (class => $value)
-> take this map and pass it to the `call` method.